tidy3d/components/simulation.py

```diff
@@ -899,6 +899,30 @@ def _validate_2d_geometry_has_2d_medium(cls, val, values):
 
         return val
 
+    @pydantic.validator("structures", always=True)
+    def _validate_advanced_material_intersections(cls, val, values):
+        """Check for unsupported intersections of advanced materials."""
+        structures = val
+        advanced_indices = []
+        advanced_structures = []
+        # first just isolate the advanced structures, to avoid unnecessary double looping
+        # keep track of indices to give helpful error message
+        for i, structure in enumerate(structures):
+            if structure.medium._is_advanced:
+                advanced_indices.append(i)
+                advanced_structures.append(structure)
+        for i, (ind1, structure_ind1) in enumerate(zip(advanced_indices, advanced_structures)):
+            for ind2, structure_ind2 in zip(
+                advanced_indices[i + 1 :], advanced_structures[i + 1 :]
+            ):
+                if not structure_ind1._compatible_with(structure_ind2):
+                    raise ValidationError(
+                        f"The structure at structures[{ind1}] and the structure at structures[{ind2}] "
```

Comment:
let's put the code bits `(structures[{ind1}])` in single quotes `'` just to make the terminal output nicer.